File: Android/API_Android/WeatherApp/src/main/java/com/example/weatherappapi/Data/WeatherData.kt
package com.example.weatherappapi.Data


data class WeatherData(
    var approvedTime: String,
    private val _data: MutableList<HourlyWeatherData> = mutableListOf(),
    private var _size: Int = 0,
    var lat: Float = 0F,
    var lon: Float = 0F
) {
    
    data class HourlyWeatherData(
        var time: String,
        var temperature: String,
        var icon: String
    )

    val data: List<HourlyWeatherData>
        get() = _data

    var size: Int
        get() = _size
        set(value) {
            _size = value
        }

    fun addHourlyData(hourlyData: HourlyWeatherData) {
        _data.add(hourlyData)
        _size++
    }
}

File: HTML5/HTML5_Dummy/weatherAppWithOutC/Data/WeatherData.js

class WeatherData {
  approvedTime; 
  size = 0;
  lat = 0;
  lon = 0;
  data = []

  getApprovedTime() {
    return this.approvedTime;
  }

  setApprovedTime(approvedTime) {
    this.approvedTime = approvedTime;
  }

  getLat() {
    return this.lat;
  }

  setLat(lat) {
    this.lat = lat;
  }

  getLon() {
    return this.lon;
  }

  setLon(lon) {
    this.lon = lon;
  }

  addHourlyData(hourlyData) {
    if (hourlyData !== null && hourlyData !== undefined) {
      this.data.push(hourlyData);
      this.size++;
    }
  }
  

  getHourlyDataSize() {
    return this.size;
  }

  getHourlyData(index) {
    if (index < 0 || index >= this.size) {
      return null;
    }
    return this.data[index];
  }
}
